modules/internal/ChapelDomain.chpl

```diff
@@ -1160,7 +1160,17 @@ module ChapelDomain {
 
     /* Return the domain map that implements this domain */
     pragma "return not owned"
-    proc dist do return _getDistribution(_value.dist);
+    proc dist {
+      use Reflection;
+      if canResolveMethod(_value, "dsiGetDist") {
```

Comment:
> Should we also check whether its return type is a record as opposed to a class?

To make sure someone doesn't accidentally write a dsiGetDist() method that returns something else?  We could, but most of our dsi*() uses are not so defensively programmed...  Note that this is a brand-new dsi method that I'm adding here, so there won't be pre-existing versions of it.  To me this is a bit like "should our `_array.this()` check that dsiAccess() returns something rather than void / returns the array's element type?  We could, but at this point in time, it feels like overkill (where interfaces are how I think we eventually want to make this stuff work).

Reply:
> Note that this is a brand-new dsi method that I'm adding here

This was the missing context for me while reviewing. I totally agree. No action needed here.